flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/matrix/DenseMatrix.java

```diff
@@ -0,0 +1,763 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.matrix;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+/**
+ * DenseMatrix stores dense matrix data and provides some methods to operate on
+ * the matrix it represents.
+ */
+public class DenseMatrix implements Serializable {
+
+	/**
+	 * Row dimension.
+	 */
+	private int m;
```

Comment:
I suppose it's more like the naming in linear algebra, where we typically say a "m*n" matrix. Since this class is more for algorithm developers who may be more familiar to mathematic, I think the naming is ok.

Reply:
Partly I agree with you, but those fields are private, and for access to it use public methods `numRows` and `numCols` and developers will be using those methods in any cases. 
For internal usage in code base more useful and informatory use common practics for developers.
Also, we can look to other linear algebra libraries, they use names related with `row` and `columns`:
https://github.com/scalanlp/breeze/blob/master/math/src/main/scala/breeze/linalg/DenseMatrix.scala#L53
https://github.com/lessthanoptimal/ejml/blob/SNAPSHOT/main/ejml-core/src/org/ejml/data/DMatrixD1.java#L39

Reply:
Partly I agree with you, but those fields are private, and for access to it use public methods `numRows` and `numCols` and external Flink developers will be using those methods in any cases. 
For internal usage in code base more useful and informatory use common practices for developers, because of the developer probably can have the gap in common math practice. Usage `row` `col` can help to avoid typos in implementation, cause in lexicographic  `m` and `n` can look very similar depends on different factors, e.g. different fonts
Also, we can look to other linear algebra libraries, they use names related with `row` and `columns`:
https://github.com/scalanlp/breeze/blob/master/math/src/main/scala/breeze/linalg/DenseMatrix.scala#L53
https://github.com/lessthanoptimal/ejml/blob/SNAPSHOT/main/ejml-core/src/org/ejml/data/DMatrixD1.java#L39

Reply:
I don't think the name matters much here. There are various libraries out there and they name it differently depending on the targeting users. 

We are essentially wrapping BLAS from `fommil/netlib`. It uses code generator to create the java classes which uses `m`/`n` rather than `row`/`col`. So I think the current naming is fine as long as the java doc is clear enough and the `m`/`n` semantic is consistent across the board.

Reply:
I got it.
Thanks @c4emmmm and @becketqin, for your comments.